guardian/typerighter
# Use Vite

## What does this change?

Use Vite, not webpack, for our client side builds. 

## Why?

- Less configuration
- Faster build times via [`swc`](https://swc.rs/)
- A significantly smaller build (>4MB, per observations in #277, to ~500kb pre-gzip)
- HMR during development:

![edit-realtime](https://user-images.githubusercontent.com/7767575/226611376-40a44258-f3a6-42fd-a451-c412e4863105.gif)

Previously, Scala was recompiling as a result of asset changes, so this is a significant improvement to the feedback loop when developing server side.

## How to test

Run locally by running `./scripts/start-manager`. The app should run locally as expected, backgrounding the build server. Try changing something in the components. The app should update in realtime.

Test the build process – which is quite different from the local dev process! – by deploying to CODE. The app should run as expected.


## How can we measure success?

A better feedback loop for devs, and a smaller bundle size in production.
